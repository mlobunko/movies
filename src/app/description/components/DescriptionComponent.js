import React from 'react';
import PropTypes from 'prop-types';

import {
  StyledDescription,
  StyledTagline,
  StyledLabel,
  StyledList,
  StyledRow,
  StyledRowRight,
  StyledRowLeft
} from './styles';

export const DescriptionComponent = ({
  title,
  tagline,
  overview,
  genres,
  productionCompanies,
  releaseDate,
  runtime,
  revenue,
  voteAvarage
}) => (
  <StyledDescription>
    <br />
    <h2>{title}</h2>
    <StyledTagline>{tagline}</StyledTagline>
    <p>{overview}</p>
    <br />
    <StyledList>
      <StyledLabel>Genres:</StyledLabel> {genres}
    </StyledList>
    <StyledList>
      <StyledLabel>Production Companies:</StyledLabel> {productionCompanies}
    </StyledList>
    <StyledRow>
      <StyledRowLeft>
        <div>
          <StyledLabel>Release Date:</StyledLabel>
        </div>
        <div>{releaseDate}</div>
      </StyledRowLeft>
      <StyledRowRight>
        <div>
          <StyledLabel>Runtime:</StyledLabel>
        </div>
        <div>{runtime} mins</div>
      </StyledRowRight>
    </StyledRow>
    <StyledRow>
      <StyledRowLeft>
        <div>
          <StyledLabel>Revenue:</StyledLabel>
        </div>
        <div>${revenue}</div>
      </StyledRowLeft>
      <StyledRowRight>
        <div>
          <StyledLabel>Vote Avarage:</StyledLabel>
        </div>
        <div>
          {voteAvarage}
          /10
        </div>
      </StyledRowRight>
    </StyledRow>
    <br />
    <br />
    <br />
  </StyledDescription>
);

DescriptionComponent.displayName = 'DescriptionComponent';

export default DescriptionComponent;

DescriptionComponent.propTypes = {
  title: PropTypes.string,
  tagline: PropTypes.string,
  overview: PropTypes.string,
  genres: PropTypes.string,
  productionCompanies: PropTypes.string,
  releaseDate: PropTypes.string,
  runtime: PropTypes.string,
  revenue: PropTypes.string,
  voteAvarage: PropTypes.string
};
