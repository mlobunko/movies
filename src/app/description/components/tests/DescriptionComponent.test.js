import React from 'react';
import { shallow } from 'enzyme';

import { DescriptionComponent } from '../DescriptionComponent';

const props = {
  title: 'family',
  tagline: 'abcd',
  overview: 'overview',
  genres: 'drama',
  productionCompanies: 'flower',
  releaseDate: '02-02-2004',
  runtime: '89',
  revenue: '222',
  voteAvarage: '4.2'
};

describe('render', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<DescriptionComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
