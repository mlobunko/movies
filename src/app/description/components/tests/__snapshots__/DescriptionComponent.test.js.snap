// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<StyledDescription>
  <br />
  <h2>
    family
  </h2>
  <StyledTagline>
    abcd
  </StyledTagline>
  <p>
    overview
  </p>
  <br />
  <StyledList>
    <StyledLabel>
      Genres:
    </StyledLabel>
     
    drama
  </StyledList>
  <StyledList>
    <StyledLabel>
      Production Companies:
    </StyledLabel>
     
    flower
  </StyledList>
  <StyledRowLeft>
    <styled.div>
      <div>
        <StyledLabel>
          Release Date:
        </StyledLabel>
      </div>
      <div>
        02-02-2004
      </div>
    </styled.div>
    <StyledRowRight>
      <div>
        <StyledLabel>
          Runtime:
        </StyledLabel>
      </div>
      <div>
        89
         mins
      </div>
    </StyledRowRight>
  </StyledRowLeft>
  <StyledRowLeft>
    <styled.div>
      <div>
        <StyledLabel>
          Revenue:
        </StyledLabel>
      </div>
      <div>
        $
        222
      </div>
    </styled.div>
    <StyledRowRight>
      <div>
        <StyledLabel>
          Vote Avarage:
        </StyledLabel>
      </div>
      <div>
        4.2
        /10
      </div>
    </StyledRowRight>
  </StyledRowLeft>
  <br />
  <br />
  <br />
</StyledDescription>
`;
